lib/Dialect/SV/SVAttributes.cpp

```diff
@@ -47,4 +44,16 @@ void SVDialect::printAttribute(Attribute attr, DialectAsmPrinter &p) const {
   llvm_unreachable("Unexpected attribute");
 }
 
-*/
+Attribute VerbatimParameterAttr::parse(MLIRContext *ctxt, DialectAsmParser &p,
+                                       Type type) {
+  StringRef verbatimString;
+  if (p.parseLess() || p.parseOptionalString(&verbatimString) ||
+      p.parseGreater())
+    return Attribute();
+
+  return VerbatimParameterAttr::get(ctxt, verbatimString);
+}
+
+void VerbatimParameterAttr::print(DialectAsmPrinter &p) const {
+  p << "verbatim<\"" << getValue() << "\">";
```

Comment:
I think this needs to escape the string, it could contain quotes right?

Reply:
Yep. I found the raw_ostream function print it escaped, but I can't find a function to un-escape it. Looks like that functionality is in MLIR's `Token` class but isn't exposed up through the parser. If there isn't another function, should I create an MLIR patch which exposes it up through the parser?

https://github.com/llvm/llvm-project/blob/205ed009a44c2b04a15aea039d8947e74856f158/mlir/lib/Parser/Token.cpp#L81